gateway/src/types.ts

```diff
@@ -34,25 +34,29 @@ export interface ApiKeyInfo {
 
 export type ProviderID = 'groq' | 'openai' | 'google-vertex' | 'anthropic' | 'test' | 'bedrock'
 // TODO | 'azure' | 'fireworks' | 'mistral' | 'cohere'
+export type APIType = 'chat' | 'responses' | 'converse' | 'anthropic' | 'gemini' | 'groq'
 
-const providerIds: Record<ProviderID, boolean> = {
-  groq: true,
-  openai: true,
-  'google-vertex': true,
+const apiTypes: Record<APIType, boolean> = {
+  chat: true,
+  responses: true,
+  converse: true,
   anthropic: true,
-  test: true,
-  bedrock: true,
+  gemini: true,
+  groq: true,
 }
 
-export const providerIdArray = Object.keys(providerIds).filter((id) => id !== 'test') as ProviderID[]
+export const APITypeArray = Object.keys(apiTypes).filter((api) => api !== 'test') as APIType[]
 
-export function guardProviderID(id: string): id is ProviderID {
-  return id in providerIds
+export function guardAPIType(api: string): api is APIType {
+  return api in apiTypes
 }
 
 export interface ProviderProxy {
   /** @providerId: decides on the logic used to process the request and response */
   providerId: ProviderID
+  /** @api: decides what API type the provider supports */
+  // TODO(Marcelo): But then we need 2 providers for OpenAI Chat & Responses!?
+  api: APIType
```

Comment:
```suggestion
  apiType: APIType
```

Just `api` is too confusing for me, I think I will change it to `apiType` here and in the console PR.